pkg/scm/git/git.go

```diff
@@ -578,3 +581,60 @@ func pipeToLog(reader io.Reader, log func(...interface{})) {
 		}
 	}
 }
+
+// CreateLocalGitDirectory creates a git directory with a commit
+func (h *stiGit) CreateLocalGitDirectory() (string, error) {
```

Comment:
Shouldn't this be part of some test API/package? Weird to have a method `CreateLocalGitDirectory` that creates a commit with an empty file.

Same for all 3 methods, aren't they better suited for some test package, and not `pkg/scm/git`?